src/app/core/services/kafka/kafka.service.ts

```diff
@@ -88,15 +93,19 @@ export class KafkaService {
       .then(([cache, headers, specifications]) => {
         const sendPromises = Object.entries(cache)
           .filter(([k]) => k)
-          .map(([k, v]: any) => {
-            const topic = this.schema.getKafkaTopic(
-              specifications,
-              v.name,
-              v.avsc
-            )
-            return this.sendToKafka(topic, k, v, headers).catch(e =>
-              this.logger.error('Failed to send data from cache to kafka', e)
-            )
+          .map(([k, val]: any) => {
+            return this.processCacheValue(val)
+              .then(v => {
+                const topic = this.schema.getKafkaTopic(
+                  specifications,
+                  v.name,
+                  v.avsc
+                )
+                return this.sendToKafka(topic, k, v, headers)
+              })
+              .catch(e =>
+                this.logger.error('Failed to send data from cache to kafka', e)
+              )
```

Comment:
We still need to process other records in the cache if some other error occurs (like sending failed due to no topic, etc). This will still stop sending of subsequent records if an error occurs, right?  I would suggest iterating through each value in the cache and catching the error on each iteration

Reply:
I've changed `getKafkaTopic()` to return a promise so that it would still send even if this fails. Previously the error in the `getKafkaTopic()` was caught in the outer promise which caused the sending of the rest of the data to be cancelled.